pkg/assets/create/creater.go

```diff
@@ -200,6 +204,75 @@ func create(ctx context.Context, manifests map[string]*unstructured.Unstructured
 	return formatErrors("failed to create some manifests", errs), reloadDiscovery
 }
 
+// createManifestAction defines the action that was performed by createManifest
+type createManifestAction int
+
+const (
+	// the action is unknown
+	unknown createManifestAction = iota
+	// the action was that the manifest was created
+	created
+	// the action was that the manifest creation was skipped
+	creationSkipped
+	// the action was the manifest was updated
+	updated
+)
+
+// createManifest creates the manifest
+// * if update is false, manifest is created or skipped if it already exists.
+// * if update is true, manifests is created if it doesn't already exist in the cluster, and patched if it already exists.
+// * if status fields are set for manifest, update is called for status subresource.
+func createManifest(ctx context.Context, resource dynamic.ResourceInterface, manifest *unstructured.Unstructured, update bool) (createManifestAction, error) {
+	incluster, action, err := createOrUpdateManifest(ctx, resource, manifest, update)
+	if err != nil {
+		return unknown, err
+	}
+
+	if _, ok := manifest.Object["status"]; !ok {
+		return action, nil
+	}
+	unstructured.SetNestedMap(incluster.Object, manifest.Object, "status")
+	incluster, err = resource.UpdateStatus(incluster, metav1.UpdateOptions{})
+	if err != nil && !kerrors.IsNotFound(err) {
+		return unknown, err
+	}
+	return action, nil
+}
+
+func createOrUpdateManifest(ctx context.Context, resource dynamic.ResourceInterface, manifest *unstructured.Unstructured, update bool) (*unstructured.Unstructured, createManifestAction, error) {
+	incluster, err := resource.Create(manifest, metav1.CreateOptions{})
+	if kerrors.IsAlreadyExists(err) {
+		incluster, err = resource.Get(manifest.GetName(), metav1.GetOptions{})
+		if err != nil {
+			return nil, unknown, err
+		}
+		if update {
+			origData, err := incluster.MarshalJSON()
+			if err != nil {
+				return nil, unknown, err
+			}
+			modData, err := manifest.MarshalJSON()
+			if err != nil {
+				return nil, unknown, err
+			}
+			patch, err := strategicpatch.CreateTwoWayMergePatch(origData, modData, manifest)
+			if err != nil {
+				return nil, unknown, err
+			}
+			incluster, err = resource.Patch(manifest.GetName(), types.MergePatchType, patch, metav1.PatchOptions{})
```

Comment:
As I recall, patching objects in this way doesn't apply the removals correctly.  What is the use-case for this?